### ISSUE TITLE
Minor warnings to fix (`-Wunused-parameter`, `-Wsign-compare`)

### ISSUE THREAD

**Issue by SuperV1234** [SFML/SFML#1785]
## Subject of the issue

There are some warnings produced when compiling SFML by source. I've looked at the code and they seem benign -- they just need to be suppressed.

## Your environment

* Windows 10, MSYS2 + MinGW
* SFML git master
* clang++ 10.x
* -Wall -Wextra -Wpedantic

## Steps to reproduce

Build.

---

```

src/SFML/Window/EglContext.cpp: In constructor ‘sf::priv::EglContext::EglContext(sf::priv::EglContext*, const sf::ContextSettings&, unsigned int, unsigned int)’:
src/SFML/Window/EglContext.cpp:163:36: warning: unused parameter ‘shared’ [-Wunused-parameter]
  163 | EglContext::EglContext(EglContext* shared, const ContextSettings& settings, unsigned int width, unsigned int height) :
      |                        ~~~~~~~~~~~~^~~~~~
src/SFML/Window/EglContext.cpp:163:67: warning: unused parameter ‘settings’ [-Wunused-parameter]
  163 | EglContext::EglContext(EglContext* shared, const ContextSettings& settings, unsigned int width, unsigned int height) :
      |                                            ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
src/SFML/Window/EglContext.cpp:163:90: warning: unused parameter ‘width’ [-Wunused-parameter]
  163 | EglContext::EglContext(EglContext* shared, const ContextSettings& settings, unsigned int width, unsigned int height) :
      |                                                                             ~~~~~~~~~~~~~^~~~~
src/SFML/Window/EglContext.cpp:163:110: warning: unused parameter ‘height’ [-Wunused-parameter]
  163 | EglContext::EglContext(EglContext* shared, const ContextSettings& settings, unsigned int width, unsigned int height) :
      |                                                                                                 ~~~~~~~~~~~~~^~~~~~



src/SFML/Window/Unix/GlxContext.cpp: In constructor ‘sf::priv::GlxContext::GlxContext(sf::priv::GlxContext*, const sf::ContextSettings&, const sf::priv::WindowImpl*, unsigned int)’:
src/SFML/Window/Unix/GlxContext.cpp:128:115: warning: unused parameter ‘bitsPerPixel’ [-Wunused-parameter]
  128 | GlxContext::GlxContext(GlxContext* shared, const ContextSettings& settings, const WindowImpl* owner, unsigned int bitsPerPixel) :



src/SFML/Window/Unix/WindowImplX11.cpp: In member function ‘void sf::priv::WindowImplX11::setVideoMode(const sf::VideoMode&)’:
src/SFML/Window/Unix/WindowImplX11.cpp:1338:33: warning: comparison of integer expressions of different signedness: ‘unsigned int’ and ‘int’ [-Wsign-compare]
 1338 |         if (res->modes[i].width == static_cast<int>(mode.width) &&
      |             ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SFML/Window/Unix/WindowImplX11.cpp:1339:34: warning: comparison of integer expressions of different signedness: ‘unsigned int’ and ‘int’ [-Wsign-compare]
 1339 |             res->modes[i].height == static_cast<int>(mode.height))
      |             ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



src/SFML/Graphics/RenderTextureImplFBO.cpp: In function ‘void {anonymous}::contextDestroyCallback(void*)’:
src/SFML/Graphics/RenderTextureImplFBO.cpp:80:39: warning: unused parameter ‘arg’ [-Wunused-parameter]
   80 |     void contextDestroyCallback(void* arg)
      |                                 ~~~~~~^~~

```

**Comment by eXpl0it3r** [comment]
Don't think we should ignore these errors

**EglContext**

https://github.com/SFML/SFML/blob/59df9d0d887f11724679f2c99dec0f05d6caeaf4/src/SFML/Window/EglContext.cpp#L163-L170

Why are we not using the provided parameters? 🤔 

**WindowImplX11**

Doesn't really cost us anything to convert to `unsigned int` then again SFML video mode is already `unsigned int`, so maybe the conversion is not needed (anymore).

**RenderTextureImplFBO**

https://github.com/SFML/SFML/blob/498d7ee79c700bba767ca42f68ced8ca116b5ada/src/SFML/Graphics/RenderTextureImplFBO.cpp#L80-L106

I guess if we don't use `arg` we could also just use `void contextDestroyCallback(void*)` which should get rid of the warning.


If anyone wants to provide PRs for these, I'm happy to accept them.

**Comment by MrZeusTheCoder** [comment]
I would also like to note that there is a unused on the Glx context as well:
```
SFML/src/SFML/Window/Unix/GlxContext.cpp:128:115: warning: unused parameter ‘bitsPerPixel’ [-Wunused-parameter]
  128 | GlxContext::GlxContext(GlxContext* shared, const ContextSettings& settings, const WindowImpl* owner, unsigned int bitsPerPixel) :
      |    
```
(Note: I am running GCC 10.1.0, not clang)

**Comment by Bromeon** [comment]
## EglContext

> Don't think we should ignore these errors
>
> https://github.com/SFML/SFML/blob/59df9d0d887f11724679f2c99dec0f05d6caeaf4/src/SFML/Window/EglContext.cpp#L163-L170
> 
> Why are we not using the provided parameters? 🤔

Good question, but it is like this [since the initial addition of the file 8 years ago](https://github.com/SFML/SFML/commit/63bbe2c91e104dda3217faf854b941b8a0fe824a#diff-ab041d39a3f86ee356da121b494b2743d607d825d97fc120d6a6f62914a8bf9e). It looks like we simply never implemented that constructor, and somehow no one bothered using it 🙂 

We have multiple options:
* Remove it -- technically a breaking change, but it's already broken, better do it at compile time than runtime.
* Implement it, might require @binary1248's or someone else's help.

## WindowImplX11

In WindowImplX11, the conversion is not needed, the X11 fields [are already `unsigned int`](https://code.woboq.org/kde/include/X11/extensions/Xrandr.h.html#_XRRModeInfo).

## Pull request

I started a draft PR (see link above this comment) to address some of the issues, but would need feedback to continue with the rest.

**Comment by binary1248** [comment]
GlContext.cpp forwards all context management functionality to the concrete platform implementations. If there are certain unused parameters on one implementation they will certainly be used by some other. As such, I don't think it makes sense to discuss cutting down on the overloads until it can be shown a certain overload isn't used by any implementation. If the aim is to reduce the warnings then I don't see why we can't just remove the parameter names.

**Comment by Bromeon** [comment]
I see. For the constructor overloads not implemented, I could at least add a `sf::err()` message, since that's SFML's current way of signaling errors.

**Comment by binary1248** [comment]
That would make sense.

**Comment by eXpl0it3r** [comment]
See #1792 but not sure if `-Wsign-conversion` is more strict than `-Wsign-compare`

**Comment by binary1248** [comment]
> See #1792 but not sure if `-Wsign-conversion` is more strict than `-Wsign-compare`

https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

```
-Wsign-compare

    Warn when a comparison between signed and unsigned values could produce an incorrect result when the signed value is converted to unsigned. In C++, this warning is also enabled by -Wall. In C, it is also enabled by -Wextra.
```
-Wsign-compare is enabled with -Wall, so #1792 will have both enabled.

**Comment by eXpl0it3r** [comment]
Implemented with #1846
